client/tests/integration/smartcontracts/query_assets_and_save_cursor/src/lib.rs

```diff
@@ -10,29 +10,43 @@ extern crate alloc;
 use alloc::string::ToString as _;
 use core::num::NonZeroU32;
 
-use iroha_smart_contract::{data_model::metadata::MetadataValueBox, parse, prelude::*};
+use iroha_smart_contract::{
+    data_model::{metadata::MetadataValueBox, query::cursor::ForwardCursor},
+    parse,
+    prelude::*,
+};
 use lol_alloc::{FreeListAllocator, LockedAllocator};
+use parity_scale_codec::{Decode, DecodeAll, Encode};
 
 #[global_allocator]
 static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
 
 getrandom::register_custom_getrandom!(iroha_smart_contract::stub_getrandom);
 
+#[derive(Debug, Decode)]
+struct QueryOutputCursor {
+    _batch: alloc::vec::Vec<Asset>,
+    cursor: ForwardCursor,
+}
+
 /// Execute [`FindAllAssets`] and save cursor to the owner's metadata.
 #[iroha_smart_contract::main]
 fn main(owner: AccountId) {
-    let asset_cursor = FindAllAssets
-        .fetch_size(FetchSize::new(Some(NonZeroU32::try_from(1).dbg_unwrap())))
-        .execute()
-        .dbg_unwrap();
-
-    let (_batch, cursor) = asset_cursor.into_parts();
+    // NOTE: Fields of QueryOutputCursor are private so we guess the layout by encoding/decoding
+    let asset_cursor = QueryOutputCursor::decode_all(
```

Comment:
This seems like a hack.. Is this something our users are supposed to to?

Reply:
no, ofc not. This is not something for users to do. This is only for this test to verify that query batching works without exposing internals of the cursor in the API. It's a dirty hack for sure

Reply:
Maybe add a comment for clarity? Integration tests can be used as examples, but some things should not be taken as an example :sweat_smile:

Reply:
done